CONTRIBUTING.md

```diff
@@ -45,22 +45,119 @@ To add a new plugin you need to do the following.
+| **imports**                | Additional Nix modules to import.                                                                                                                                                                            | No       | `[]`            |
+| **description**            | A brief description of the plugin.                                                                                                                                                                           | No       | `null`          |
+| **deprecateExtraOptions**  | Indicating whether to deprecate the `extraOptions` attribute. Mainly used for old plugins.                                                                                                                   | No       | `false`         |
+| **optionsRenamedToSettings** | Options that have been renamed and move to the `settings` attribute.                                                                                                                                       | No       | `[]`            |
+| **isColorscheme**          | Indicating whether the plugin is a colorscheme.                                                                                                                                                              | No       | `false`         |
+| **colorscheme**            | The name of the colorscheme.                                                                                                                                                                                 | No       | `name` parameter |
+| **configLocation**         | The location for the Lua configuration.                                                                                                                                                                      | No       | `"extraConfigLuaPre"` if `isColorscheme` is `true`, otherwise `"extraConfigLua"` |
+| **hasConfigAttrs**         | Indicating whether the plugin has configuration attributes.                                                                                                                                                  | No       | `true`          |
+| **originalName**           | The original name of the plugin.                                                                                                                                                                             | No       | `name` parameter |
+| **settingsOptions**        | Options for the plugin's settings.                                                                                                                                                                           | No       | `{}`            |
```

Comment:
```suggestion
| **settingsOptions**        | Options representing the plugin's settings. This is optional because `settings` is a "freeform" option. See <insert link here>.                                                                                                                                                                           | No       | `{}`            |
```

We want to keep reminding contributors that it is not necessary to represent plugin settings as nix options in order for end users to configure them.

We should link to the relevant sub-heading where this is discussed in more detail.

Reply:
I can't a relevant link to set here for a simple freeform plugin already added to nixvim. Do you have one?

Reply:
I think [`#declaring-plugin-options`](https://github.com/nix-community/nixvim/blob/main/CONTRIBUTING.md#declaring-plugin-options) is probably what I had in mind.